### PR TITLE
Improves pAI code

### LINKED ISSUE #22286: pAI Pushing
You are able to push people on a scooter as a pAI, vs walking through everyone else.

### LINKED ISSUE #23224: PAIS ARE FUCKING HARD TO STOP
i know people say ion gun but i mean those things dont even get fucking stunned i blame mekhi there was a PAI in a emagged janibot this round i destroyed the janibot it kept runnin runnin and runnin i tried to stun it with a baton but it didnt even fucking stop I MEAN DRONES CAN BE STUNNED WHY CANT PAI'S???

### LINKED ISSUE #28127: Pai's can buckle people to beds etc
And this was just the pocket version

### LINKED ISSUE #28707: Bug With pAI Suicide
I've tested this twice now and believe it's a consistent problem, though I'm not sure if it only applies in this specific condition or if it's a sweeping problem with pAIs, BUT.

Upon using the pAI Suicide function of a pAI's special tab while in holochassis form on the ground (and in space, tested) the user is reverted to ghost form, however the handheld pAI device despawns as well and completely disappears from the game. It doesn't function this way when a pAI is forcibly wiped by a user in the pAI interaction interface so I can only assume this must be a bug.

### PR DESCRIPTION
**Pull request by ghost**
**warning this is very likely to be shitcode**
god help me, more shitcode coming soon 🥉 

Fixes #28707
Fixes #28127
Fixes #22286

:cl: Frozenguy5/PraiseRatvar
fix: pAI can no longer buckle themselves or people to anything.
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
You monster :cry:

File: code/game/machinery/computer/robot.dm [hunk 1]
```diff
@@ -170,6 +182,18 @@
 				s.start()
 				D.visible_message("<span class='danger'>\the [D] self destructs!</span>")
 				D.gib()
+				
+	else if (href_list["killpai"])
+		if(src.allowed(usr))
```
Comment (Cyberboss) [inline code/game/machinery/computer/robot.dm]: remove src

File: code/game/machinery/computer/robot.dm [hunk 2]
```diff
@@ -170,6 +182,18 @@
...
+				
+	else if (href_list["killpai"])
+		if(src.allowed(usr))
+			var/mob/living/simple_animal/drone/P = locate(href_list["killpai"]) in GLOB.mob_list
+				var/turf/T = get_turf(P)
+				message_admins("[ADMIN_LOOKUPFLW(usr)] detonated [key_name_admin(P)][ADMIN_JMP(T)]!")
+				log_game("[key_name(usr)] detonated [key_name(P)]!")
+				var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+				s.set_up(3, 1, P)
+				s.start()
```
Comment (Cyberboss) [inline code/game/machinery/computer/robot.dm]: use do_sparks

File: code/modules/client/verbs/suicide.dm [hunk 3]
```diff
@@ -114,17 +114,20 @@
...
 
 /mob/living/silicon/pai/verb/suicide()
-	set category = "pAI Commands"
-	set desc = "Kill yourself and become a ghost (You will receive a confirmation prompt)"
-	set name = "pAI Suicide"
-	var/answer = input("REALLY kill yourself? This action can't be undone.", "Suicide", "No") in list ("Yes", "No")
-	if(answer == "Yes")
-		var/turf/T = get_turf(src.loc)
-		T.visible_message("<span class='notice'>[src] flashes a message across its screen, \"Wiping core files. Please acquire a new personality to continue using pAI device functions.\"</span>", null, \
+    set hidden = 1
```
Comment (Cyberboss) [inline code/modules/client/verbs/suicide.dm]: TRUE

File: code/modules/client/verbs/suicide.dm [hunk 4]
```diff
@@ -114,17 +114,20 @@
...
+	if(!canSuicide())
+		return
+	var/confirm = alert("Are you sure you want to wipe yourself?", "Confirm data wipe", "Yes", "No")
+	if(!canSuicide())
+		return
+	if(confirm == "Yes")
+		suiciding = 1
```
Comment (Cyberboss) [inline code/modules/client/verbs/suicide.dm]: TRUE

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 5]
```diff
@@ -76,6 +76,12 @@
 /mob/living/silicon/pai/movement_delay()
 	. = ..()
 	. += slowdown
+	
+/mob/living/silicon/pai/can_unbuckle()
+	return 0
```
Comment (Cyberboss) [inline code/modules/mob/living/silicon/pai/pai.dm]: FALSE

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 6]
```diff
@@ -76,6 +76,12 @@
 /mob/living/silicon/pai/movement_delay()
 	. = ..()
 	. += slowdown
+	
+/mob/living/silicon/pai/can_unbuckle()
+	return 0
+
+/mob/living/silicon/pai/can_buckle()
+    return 0
```
Comment (Cyberboss) [inline code/modules/mob/living/silicon/pai/pai.dm]: FALSE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by PraiseRatvar** [comment]
done all requested changes

**Comment by kevinz000** [comment]
balance change @KorPhaeron 
also pais are weak as fuck they're slow can't ventcrawl have no access and basically can't do anything
why the hell do they need to be on this?

File: code/game/machinery/computer/robot.dm [hunk 7]
```diff
@@ -87,6 +87,18 @@
 
 	if(!drones)
 		dat += "No Drone Units detected within access parameters."
+
```
Comment (KorPhaeron) [inline code/game/machinery/computer/robot.dm]: This violates the freeze

**Review by imtakingabreakdontatme — DISMISSED** [review]

**Comment by KorPhaeron** [comment]
Removed the remote wiping changes because they violate the freeze

**Comment by PraiseRatvar** [comment]
alrighty

File: code/modules/client/verbs/suicide.dm [hunk 8]
```diff
@@ -114,17 +114,18 @@
...
-	set desc = "Kill yourself and become a ghost (You will receive a confirmation prompt)"
-	set name = "pAI Suicide"
-	var/answer = input("REALLY kill yourself? This action can't be undone.", "Suicide", "No") in list ("Yes", "No")
-	if(answer == "Yes")
-		var/turf/T = get_turf(src.loc)
-		T.visible_message("<span class='notice'>[src] flashes a message across its screen, \"Wiping core files. Please acquire a new personality to continue using pAI device functions.\"</span>", null, \
+    set hidden = TRUE
+		return
+	var/confirm = alert("Are you sure you want to wipe yourself?", "Confirm data wipe", "Yes", "No")
+		return
```
Comment (Cyberboss) [inline code/modules/client/verbs/suicide.dm]: Uhhhhh

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/client/verbs/suicide.dm (hunk 8 above)
Comment (optimumtact) [inline code/modules/client/verbs/suicide.dm]: *return*

File: code/game/machinery/computer/robot.dm [hunk 9]
```diff
@@ -170,7 +170,7 @@
 				s.start()
 				D.visible_message("<span class='danger'>\the [D] self destructs!</span>")
 				D.gib()
-
-
+
```
Comment (optimumtact) [inline code/game/machinery/computer/robot.dm]: can you delete all this random whitespace you're adding

File: code/modules/client/verbs/suicide.dm [hunk 10]
```diff
@@ -114,17 +114,18 @@
...
 
 /mob/living/silicon/pai/verb/suicide()
-	set category = "pAI Commands"
-	set desc = "Kill yourself and become a ghost (You will receive a confirmation prompt)"
-	set name = "pAI Suicide"
-	var/answer = input("REALLY kill yourself? This action can't be undone.", "Suicide", "No") in list ("Yes", "No")
-	if(answer == "Yes")
-		var/turf/T = get_turf(src.loc)
-		T.visible_message("<span class='notice'>[src] flashes a message across its screen, \"Wiping core files. Please acquire a new personality to continue using pAI device functions.\"</span>", null, \
+    set hidden = TRUE
```
Comment (optimumtact) [inline code/modules/client/verbs/suicide.dm]: this is not needed

**Comment by optimumtact** [comment]
Stale, ask for reopen or make a new pr when you're ready to continue

**Pull request closed without merging**